runners/spark/src/main/java/org/apache/beam/runners/spark/translation/SparkContextFactory.java

```diff
@@ -93,6 +94,9 @@ private static JavaSparkContext createSparkContext(SparkContextOptions contextOp
         // set master if not set.
         conf.setMaster(contextOptions.getSparkMaster());
       }
+      if (!conf.contains("spark.jars") && !Strings.isNullOrEmpty(contextOptions.getJars())) {
```

Comment:
Wouldn't it be better to detect the files automatically using something like DataflowRunner#detectClassPathResourcesToStage:
https://github.com/apache/beam/blob/master/runners/google-cloud-dataflow-java/src/main/java/org/apache/beam/runners/dataflow/DataflowRunner.java#L1181